src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb

```diff
@@ -2185,6 +2185,50 @@ class MyClass
             End Using
         End Function
 
+        <WpfFact>
+        Friend Async Function TestCompilerAnalyzerForSpanBasedQuery() As Task
+            Dim test = <Workspace>
+                           <Project Language="C#" CommonReferences="true">
+                               <Document><![CDATA[
+class C
+{
+    void M1()
+    {
+        int x1 = 0;
+    }
+}]]>
+                               </Document>
+                           </Project>
+                       </Workspace>
+
+            Using workspace = TestWorkspace.CreateWorkspace(test, composition:=s_compositionWithMockDiagnosticUpdateSourceRegistrationService)
+                Dim solution = workspace.CurrentSolution
+                Dim project = solution.Projects.Single()
+
+                ' Add compiler analyzer
+                Dim analyzer = DiagnosticExtensions.GetCompilerDiagnosticAnalyzer(LanguageNames.CSharp)
+                Dim analyzerReference = New AnalyzerImageReference(ImmutableArray.Create(analyzer))
+                project = project.AddAnalyzerReference(analyzerReference)
+
+                ' Get span to analyze
+                Dim document = project.Documents.Single()
+                Dim root = Await document.GetSyntaxRootAsync(CancellationToken.None)
+                Dim localDecl = root.DescendantNodes().OfType(Of CodeAnalysis.CSharp.Syntax.LocalDeclarationStatementSyntax).Single()
+                Dim span = localDecl.Span
+
+                Dim mefExportProvider = DirectCast(workspace.Services.HostServices, IMefHostExportProvider)
+                Assert.IsType(Of MockDiagnosticUpdateSourceRegistrationService)(workspace.GetService(Of IDiagnosticUpdateSourceRegistrationService)())
+                Dim diagnosticService = Assert.IsType(Of DiagnosticAnalyzerService)(workspace.GetService(Of IDiagnosticAnalyzerService)())
+                Dim incrementalAnalyzer = diagnosticService.CreateIncrementalAnalyzer(workspace)
+
+                ' Verify diagnostics for span
+                Dim diagnostics As New PooledObjects.ArrayBuilder(Of DiagnosticData)
+                Await diagnosticService.TryAppendDiagnosticsForSpanAsync(document, span, diagnostics)
+                Dim diagnostic = Assert.Single(diagnostics)
+                Assert.Equal("CS0219", diagnostic.Id)
+            End Using
+        End Function
```

Comment:
similar point about testing different spans.

Reply:
@CyrusNajmabadi Adding the IDE test for testing different spans failed as the IDE seems to be querying the compiler diagnostics for entire containing member instead of just the line span: https://github.com/dotnet/roslyn/blob/4fc72457f6c60c0f55ddaa09b0a485e0c425cefc/src/Features/Core/Portable/Diagnostics/DocumentAnalysisExecutor.cs#L251-L284

This adjusted span workaround seems to be added for #1557, which is pretty old and has no failing tests or scenarios. I am going to just remove that adjusted span computation code and address any fallout subsequently based on failing scenarios.

Reply:
Works for me.

Reply:
So attempting to remove `GetAdjustedSpanForCompilerAnalyzerAsync` workaround leads to few [integration test failures](https://dev.azure.com/dnceng/public/_build/results?buildId=1394897&view=ms.vss-test-web.build-test-results-tab&runId=40546354&paneView=debug&resultId=100274). So the claim in #1557 is indeed true - invoking `SemanticModel.GetDiagnostics(lineSpan)` with just the current line span instead of entire method body span indeed does not return all the diagnostics for the current line. I have reverted the commit that removed the workaround and instead adjusted the above test to also account for this fact.

 @333fred @jcouv - #1557 does indeed represent a potential perf improvement case for light bulb - IDE is currently being forced to always fetch compiler semantic diagnostics for entire method body span instead of line span due to this bug. If compiler team were to fix this, IDE can remove this workaround and see a perf improvement by requesting the diagnostics just for the current line span.